src/lambdas/breeds-get.ts

```diff
@@ -0,0 +1,22 @@
+import { getDogBreeds } from '../services/http-req/dogBreedRequest'
+import { flattenObjectValueKeyOrder } from '../services/utilities/flattenObjectValueKeyOrder'
+import { RequestedDogBreeds, ErrorResponse, SuccessResponse } from './types'
+
+export const getBreeds = async (): Promise<SuccessResponse<string[]> | ErrorResponse> => {
+  try {
+    const apiData: RequestedDogBreeds = await getDogBreeds()
+    if (apiData.status === 'error') {
+      throw new Error('Something went wrong')
+    }
+    const payload: string[] = flattenObjectValueKeyOrder(apiData.data)
+    return {
+      statusCode: 200,
+      body: payload,
+    }
+  } catch (err) {
+    return {
+      statusCode: 500,
+      message: err.message,
+    }
+  }
```

Comment:
If a timeout were to occur, could we return a more specific response and still handle the generic 500 errors as well?

Reply:
We also want to be able to keep `err` typed as unknown since we do not know the type of error that was thrown. Is there a way we can ensure the error thrown was related to a timeout?

Reply:
Thanks for pointing this out. This implementation is definitely too brittle. I decided to move the logic for generating ErrorResponse objects to a helper function where different error messages and response codes can be set depending on the error name.